python/lsst/ts/ess/csc/data_client/siglent_ssa3000x_spectrum_analyzer_data_client.py

```diff
@@ -193,70 +204,53 @@ async def write(self, data: str) -> None:
-            if self.connected and self.simulation_mode == 0:
-                # Make sure that the correct frequency range is used by the
-                # spectrum analyzer.
-                await self.write(self.set_freq_start_cmd)
-                await self.write(self.set_freq_stop_cmd)
-            while self.connected:
-                timestamp = utils.current_tai()
-                await self.write(self.query_trace_data_cmd)
-                assert self.client is not None  # make mypy happy
-                read_bytes = await asyncio.wait_for(
-                    self.client.readuntil(TERMINATOR),
-                    timeout=self.config.read_timeout,
-                )
-                self.num_consecutive_read_timeouts = 0
-                raw_data = read_bytes.decode().strip()
-                raw_data_items = raw_data.split(",")
-                data = [float(i.strip()) for i in raw_data_items]
-                if len(data) < EXPECTED_NUMBER_OF_DATA_POINTS and not self.read_data:
-                    logging.warning(
-                        f"Data of length {len(data)} read. Ignoring because this is "
-                        f"read #{self.number_of_truncated_data_read} out of {self.config.max_read_timeouts}."
-                    )
-                    self.read_data = True
-                elif len(data) != EXPECTED_NUMBER_OF_DATA_POINTS:
-                    raise RuntimeError(
-                        f"Encountered {len(data)} data points instead of "
-                        f"{EXPECTED_NUMBER_OF_DATA_POINTS}. Check the Spectrum "
-                        f"Analyzer and the configuration."
-                    )
-                else:
-                    try:
-                        await self.topics.tel_spectrumAnalyzer.set_write(
-                            startFrequency=self.start_frequency,
-                            stopFrequency=self.stop_frequency,
-                            spectrum=data,
-                            timestamp=timestamp,
-                        )
-                    except Exception as e:
-                        self.log.exception(f"Failed to handle {data=}: {e!r}")
-
-                # Maybe a bit of an overkill but this ensures that no drift
-                # gets introduced while sleeping.
-                sleep_delay = (
-                    utils.current_tai() - timestamp - self.config.poll_interval
-                )
-                if sleep_delay > 0:
-                    await asyncio.sleep(sleep_delay)
+            read_bytes = await asyncio.wait_for(
+                self.client.readuntil(TERMINATOR),
+                timeout=self.config.read_timeout,
+            )
         except asyncio.TimeoutError:
-            self.num_consecutive_read_timeouts += 1
-            self.log.warning(
-                f"Read timed out. This is timeout #{self.num_consecutive_read_timeouts} "
-                f"of {self.config.max_read_timeouts} allowed."
+            self._have_seen_data = False
```

Comment:
Consider generalizing the `except` to catch I/O errors, as well (e.g. `except Exception`, which is probably what I'd do, or a list of possible exceptions).